vulnfeeds/cpp/main.go

```diff
@@ -167,65 +284,112 @@ func main() {
 
 	for _, cve := range parsed.CVEItems {
 		refs := cve.CVE.References.ReferenceData
-		patchRefs := 0
-		cpes := cves.CPEs(cve)
+		patchRefCount := 0
+		CPEs := cves.CPEs(cve)
+		repos := make(map[string]string)
 
-		if len(refs) == 0 && len(cpes) == 0 {
+		if len(refs) == 0 && len(CPEs) == 0 {
 			Logger.Infof("FYI: skipping %s due to:", cve.CVE.CVEDataMeta.ID)
 			Logger.Infof("\t * lack of CPEs")
 			Logger.Infof("\t * lack of references")
 			continue
 		}
 
 		// Does it have any application CPEs?
-		appCpes := 0
-		for _, cpeStr := range cves.CPEs(cve) {
-			cpe, ok := cves.ParseCPE(cpeStr)
+		appCPECount := 0
+		for _, CPEstr := range cves.CPEs(cve) {
+			CPE, ok := cves.ParseCPE(CPEstr)
 			if ok {
-				if cpe.Part == "a" {
-					appCpes += 1
+				if CPE.Part == "a" {
+					appCPECount += 1
 				}
 			} else {
-				Logger.Fatalf("Failed to parse CPE %s: %v", cpeStr, err)
+				Logger.Fatalf("Failed to parse CPE %s: %v", CPEstr, err)
 			}
 		}
 
-		if appCpes == 0 {
-			Logger.Infof("FYI: skipping %s due to:", cve.CVE.CVEDataMeta.ID)
-			Logger.Infof("\t * believed non-software")
+		if appCPECount == 0 {
+			// Not software, skip.
 			continue
 		}
 
 		Logger.Infof("%s", cve.CVE.CVEDataMeta.ID)
 		for _, ref := range refs {
 			// Are any of the reference's tags 'Patch'?
 			for _, tag := range ref.Tags {
-				// TODO(apollock): determine rate of false negatives
-				if tag == "Patch" && IsRepoURL(ref.URL) {
+				if tag == "Third Party Advisory" {
+					continue
+				}
+				// Alternative to the above:
+				// if tag == "Patch" && IsRepoURL(ref.URL) {
+				if IsRepoURL(ref.URL) {
 					Logger.Infof("\t * %s", ref.URL)
-					patchRefs += 1
+					// CVE entries have one set of references, but can have multiple CPEs
```

Comment:
This is a lot of nesting and getting hard to read.. Consider putting some of this stuff (either the code here or the entire for loop) into its own function.